src/intersections/rayline.jl

```diff
@@ -0,0 +1,33 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+#=
+The intersection type can be one of four types:
+
+1. intersect at one inner point (CrossingRayLine -> Point)
+2. intersect at origin of ray (TouchingRayLine -> Point)
+3. overlap of line and ray (OverlappingRayLine -> Ray)
+4. do not overlap nor intersect (NoIntersection)
+=#
+function intersection(f, r1::Ray{N,T}, l1::Line{N,T}) where {N,T}
```

Comment:
can we get rid of the "1" suffix in the arguments? We only have one ray and one line in this case.

Reply:
well I would keep it, since `r` is already used for the rank.
Letter + number is a good convention to label primitives or polytopes.
I have done the same in `raysegments.jl`

Reply:
The clash with the rank variable is a good reason to keep the 1 suffix here. It is still strange, but no big deal.